### PR TITLE
Change max_port_file_count to warn rather than failing to calculate an ABI hash.

### PR DESCRIPTION
**Pull request by BillyONeal**
Resolves https://github.com/microsoft/vcpkg/issues/31975 .

### CODE REVIEW

**Review by JavierMatosD — APPROVED** [review]

**Comment by ras0219-msft** [comment]
The reason for this max file count was to avoid appearing to "hang" if users accidentally cloned source repos into the port directory. We will recurse into .git directories and attempt to hash tens or hundreds of thousands of files.

Would it be more appropriate to raise this limit to 1,000 instead of removing it entirely?

**Comment by BillyONeal** [comment]
> to avoid appearing to "hang"

I think we should address this with a warning rather than just refusing to calculate the correct cache information. I'll restore the limit but make it trigger a warning rather than refusing to generate a cache entry.

File: include/vcpkg/base/message-data.inc.h [hunk 1]
```diff
@@ -1129,6 +1129,13 @@ DECLARE_MESSAGE(HashFileFailureToRead,
                 (msg::path),
                 "Printed after ErrorMessage and before the specific failing filesystem operation (like file not found)",
                 "failed to read file \"{path}\" for hashing: ")
+DECLARE_MESSAGE(
+    HashPortManyFiles,
+    (msg::package_name, msg::count),
+    "",
+    "The {package_name} contains a {count} files. Hashing these contents may take a long time when determining the ABI "
```
Comment (ras0219-msft) [inline include/vcpkg/base/message-data.inc.h:1136]:
```suggestion
    "The {package_name} contains {count} files. Hashing these contents may take a long time when determining the ABI "
```

File: include/vcpkg/base/message-data.inc.h [hunk 2]
```diff
@@ -1129,6 +1129,13 @@ DECLARE_MESSAGE(HashFileFailureToRead,
...
+    "hash for binary caching. You should probably consider reducing the number of files. Common causes of this are "
```
Comment (ras0219-msft) [inline include/vcpkg/base/message-data.inc.h:1137]:
```suggestion
    "hash for binary caching. Consider reducing the number of files. Common causes of this are "
```

File: include/vcpkg/base/message-data.inc.h [hunk 3]
```diff
@@ -1129,6 +1129,13 @@ DECLARE_MESSAGE(HashFileFailureToRead,
...
+    "accidentally checking out source or build trees into a port's directory.")
```
Comment (ras0219-msft) [inline include/vcpkg/base/message-data.inc.h:1138]:
```suggestion
    "accidentally checking out source or build files into a port's directory.")
```

**Review by ras0219-msft — APPROVED** [review]

**Pull request merged**